force-app/main/default/classes/AFFL_AccRecordType_TEST.cls

```diff
@@ -189,6 +208,81 @@ public with sharing class AFFL_AccRecordType_TEST {
         updateAdminAccRecordTypeBulk();
     }
 
+    /*
+     * Ensure redundant affiliaition RT mappings is surfaced as an error upon contact update.
+     * Test Scenario: T-3939010
+     */
+    @isTest
+    private static void invalidRedundantAffiliationRTMappingCheck_FieldLabels() {
+        setupWithFieldLabels();
+        // Insert redundant affiliation mapping
+        insert new Affl_Mappings__c(Name = 'Household Account', Account_Record_Type__c = 'HH_Account');
+
+        try {
+            changePrimaryAffiliationRecordType();
+        } catch (Exception ex) {
+            System.assert(ex.getMessage().contains('Row with duplicate Name'));
+        }
+    }
+
+    /*
+     * Ensure redundant affiliaition field mappings is surfaced as an error upon contact update.
+     * Test Scenario: T-3939011
+     */
+    @isTest
+    private static void invalidRedundantAffiliationFieldMappingCheck_FieldLabels() {
+        setupWithFieldLabels();
+        // Insert redundant affiliation mapping
+        insert new Affl_Mappings__c(
+            Name = 'Household Account',
+            Primary_Affl_Field__c = 'Primary Business Organization'
+        );
+
+        try {
+            changePrimaryAffiliationRecordType();
+        } catch (Exception ex) {
+            System.assert(ex.getMessage().contains('Row with duplicate Name'));
+        }
+    }
+
+    /*
+     * Ensure affiliaition mappings consisting of disabled record types appropriately surface as an error.
+     * Test Scenario: T-3939012
+     */
+    @isTest
+    private static void disabledAccountRTAffiliationMappingCheck_FieldLabels() {
+        setupWithFieldLabels();
+        Contact contact = UTIL_UnitTestData_API.getContact();
+        insert contact;
+
+        Account bizOrg = UTIL_UnitTestData_API.getMultipleTestAccounts(1, orgRecTypeID)[0];
+        insert bizOrg;
+
+        Affiliation__c bizAffl = new Affiliation__c(Contact__c = contact.Id, Account__c = bizOrg.Id, Primary__c = true);
+
+        TDTM_ProcessControl.setRecursionFlag(
+            TDTM_ProcessControl.registeredTrigger.AFFL_MultiRecordType_TDTM_Before_Insert,
+            false
+        );
+
+        Test.startTest();
+        insert bizAffl;
+        Test.stopTest();
+        contact = [SELECT AccountId, Primary_Household__c, Primary_Organization__c FROM Contact WHERE Id = :contact.Id];
+
+        // Given BizOrg RT disabled in org
+        Integer isBizAcctRTDisabledInOrg = [
+            SELECT COUNT()
+            FROM RecordType
+            WHERE DeveloperName = 'Business_Organization' AND SObjectType = 'Account' AND IsActive = FALSE
+        ];
+
+        //Verify the primary business organization field is not populated when the RT is disabled
+        if (isBizAcctRTDisabledInOrg > 0) {
+            System.assertEquals(null, contact.Primary_Organization__c);
+        }
+    }
```

Comment:
Not a necessary change, but why not frontload the check that would make this entire test valid? This allows anyone viewing the test to immediately determine when the test is or isn't valid rather than placing the condition at the end.

```suggestion
        // Given BizOrg RT disabled in org
        Integer isBizAcctRTDisabledInOrg = [
            SELECT COUNT()
            FROM RecordType
            WHERE DeveloperName = 'Business_Organization' AND SObjectType = 'Account' AND IsActive = FALSE
        ];

        //Verify the primary business organization field is not populated when the RT is disabled
        if (isBizAcctRTDisabledInOrg == 0) {
            return;
        }

        setupWithFieldLabels();
        Contact contact = UTIL_UnitTestData_API.getContact();
        insert contact;

        Account bizOrg = UTIL_UnitTestData_API.getMultipleTestAccounts(1, orgRecTypeID)[0];
        insert bizOrg;

        Affiliation__c bizAffl = new Affiliation__c(Contact__c = contact.Id, Account__c = bizOrg.Id, Primary__c = true);

        TDTM_ProcessControl.setRecursionFlag(
            TDTM_ProcessControl.registeredTrigger.AFFL_MultiRecordType_TDTM_Before_Insert,
            false
        );

        Test.startTest();
        insert bizAffl;
        Test.stopTest();
        contact = [SELECT AccountId, Primary_Household__c, Primary_Organization__c FROM Contact WHERE Id = :contact.Id];

        System.assertEquals(null, contact.Primary_Organization__c);
    }
```